### PR TITLE
[1.9] Update references to AutoMaterializePolicy in docs

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

Removes / updates any references to AutoMaterializePolicy in the docs

## Changelog

> Insert changelog entry or delete this section.

### CODE REVIEW

**Comment by OwenKephart** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25520?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25520** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25520?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#25509** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25509?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25513** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25513?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25512** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25512?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @OwenKephart and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/content/guides/dagster/managing-ml.mdx [hunk 1]
```diff
@@ -28,23 +28,21 @@ You might have thought about your data sources, feature sets, and the best model
...
 
-@asset(
-    auto_materialize_policy=AutoMaterializePolicy.eager(),
-)
+@asset(automation_condition=AutomationCondition.eager())
 def my_ml_model(my_data): ...
 ```
 
-Some machine learning models might more be cumbersome to retrain; it also might be less important to update them as soon as new data arrives. For this, a lazy auto-materialization policy which can be used in two different ways. The first, by using it with a `freshness_policy` as shown below. In this case, `my_ml_model` will only be auto-materialized once a week.
+Some machine learning models might more be cumbersome to retrain; it also might be less important to update them as soon as new data arrives. For this, the `on_cron` condition may be used, which will cause the asset to be updated on a given cron schedule, but only after all of its upstream dependencies have been updated.
```
Comment (cmpadden) [inline docs/content/guides/dagster/managing-ml.mdx:45]:
```suggestion
Some machine learning models might be more cumbersome to retrain; it also might be less important to update them as soon as new data arrives. For this, the `on_cron` condition may be used, which will cause the asset to be updated on a given cron schedule, but only after all of its upstream dependencies have been updated.
```

**Review by cmpadden — APPROVED** [review]
Approved pending green BK, looks good!

**Comment by OwenKephart** [comment]
### Merge activity

* **Oct 24, 6:46 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/25520).

**Pull request merged**